### DISCUSSION TITLE
Changing PD gains of stiffness and damping for torque control

### DISCUSSION THREAD

**Discussion by tesollo-admin** [isaac-sim/IsaacLab#531] [Q&A]
Hi, @Mayankm96 .

Thank you for your big help.

I have some related questions about changing the mode in a simulation.

Currently, I am running a simulation with my own imported robot gripper and would like to control the joints using torque (effort control). In my case, I want to use the torque control mode only during the grasping moment. So, I tried to set the values of stiffness and damping to zero (the initial setting is for position control). However, it doesn't work with the following code:
```javascript
  set_gain_p = torch.tensor([[0.0, 0.0, 0.0]], device='cuda:0')
  set_gain_d = torch.tensor([[0.0, 0.0, 0.0]], device='cuda:0')
  robot.actuators["f_joint_3"].stiffness = set_gain_p
  robot.actuators["f_joint_4"].stiffness = set_gain_p
  robot.actuators["f_joint_3"].damping = set_gain_d
  robot.actuators["f_joint_4"].damping = set_gain_d
```
Also have a problem with zero PD gains at the initial setting. The joints do not control with the generated torques.

In Isaac Sim, I checked that the following works well:
```javascript
switch_control_mode("effort", joint_indices=ids_gripper)
set_joint_efforts(efforts=torque_values, joint_indices=ids_gripper)
```

Is there any solution?

Thank you in advance.

**Comment by tesollo-admin** [comment]
Solved with `set_joint_effort_target`, `write_joint_stiffness_to_sim`, `write_joint_damping_to_sim`

Thanks

**Reply by jojolin329** [reply]
Hi, I’m experiencing a similar issue — I’d like to use position control for one task and then switch to torque control afterward. Would it be possible to share how you set up your robot configuration? For example, which actuator configuration are you using — Implicit or Explicit? Also, did you refer to any specific documentation or code examples when setting up your robot?

From the source code, it seems that when using the implicit actuator configuration, the write_data_to_sim function only sends the desired position and velocity to PhysX, and an internal built-in PD position controller is applied. I’m not sure whether setting the damping and stiffness gains to zero and then switching to torque control using a command like set_joint_effort_target would actually work.

I would really appreciate it if you could help with your experience.
